parser.go

```diff
@@ -329,6 +340,19 @@ func (r *Rule) option(key item, l *lexer) error {
 			return fmt.Errorf("invalid reference definition: %s", refs)
 		}
 		r.References = append(r.References, &Reference{Type: refs[0], Value: refs[1]})
+	case "metadata":
+		nextItem := l.nextItem()
+		if nextItem.typ != itemOptionValue {
+			return errors.New("no valid value for metadata")
+		}
+		metas := metaSplitRE.Split(nextItem.value, -1)
+		for _,kv := range metas{
+			meta_tmp := strings.SplitN(kv, " ", 2)
+			if len(meta_tmp) != 2 {
+				return fmt.Errorf("invalid metadata definition: %s", meta_tmp)
+			}
+			r.Metas = append(r.Metas, &Metadata{Key: meta_tmp[0], Value: meta_tmp[1]})
```

Comment:
apply strings.TrimSpace to the key and value, and I think we're good!

Reply:
In suricata rules,there is no need to use TrimSpace,but I am not familiar with other nids rules, so I add TrimSpace function just in case.

Reply:
Thanks, yeah, spaces are fine, but I think they're usually typos. Thanks for changing!